snappy/install.go

```diff
@@ -46,10 +46,44 @@ const (
 	AllowOEM
 )
 
-// Update the installed snappy packages, it returns the updated Parts
+func doUpdate(part Part, flags InstallFlags, meter progress.Meter) error {
+	if _, err := part.Install(meter, flags); err == ErrSideLoaded {
+		logger.Noticef("Skipping sideloaded package: %s", part.Name())
+		return nil
+	} else if err != nil {
+		return err
+	}
+	if err := GarbageCollect(part.Name(), flags, meter); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Update updates the selected name
+func Update(name string, flags InstallFlags, meter progress.Meter) ([]Part, error) {
```

Comment:
Well, yes deduping is what I meant. But also that you compare the two, as they were different: yours loads Installed() before ListUpdates. I presume this is to have the 'not installed' error message, and if that's the case, then it's fine; otherwise, it's unnecessary and shouldn't be done imo.